include/zephyr/drivers/comparator.h

```diff
@@ -0,0 +1,201 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_DRIVERS_COMPARATOR_H_
+#define ZEPHYR_INCLUDE_DRIVERS_COMPARATOR_H_
+
+/**
+ * @brief Comparator Driver APIs
+ * @defgroup comparator_interface Comparator Driver APIs
+ * @ingroup io_interfaces
+ * @{
+ */
+
+#include <zephyr/device.h>
+#include <zephyr/dt-bindings/comparator/comparator.h>
+#include <zephyr/kernel.h>
+#include <zephyr/sys/util_macro.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @name Comparator state flags
+ * @{
+ */
+
+/** Positive input voltage is below the negative input voltage. */
+#define COMPARATOR_STATE_BELOW BIT(0)
+/** Positive input voltage is above the negative input voltage. */
+#define COMPARATOR_STATE_ABOVE BIT(1)
```

Comment:
I see there is no cross state - ```COMPARATOR_STATE_CROSS``` . Could be useful to test for example power lines if sine wave is correct